src/PowerShellEditorServices/Extensions/EngineExtensions/EditorEngine.cs

```diff
@@ -0,0 +1,73 @@
+﻿
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.PowerShell.EditorServices.Utility;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using System;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Microsoft.PowerShell.EditorServices.Extensions
+{
+    public class EditorEngine
+    {
+        private static readonly Assembly s_psesAsm = typeof(EditorEngine).Assembly;
+
+        private static readonly Lazy<Func<IDisposable>> s_enterPsesReflectionContextLazy = new Lazy<Func<IDisposable>>(GetPsesAlcReflectionContextEntryFunc);
+
+        private static Func<IDisposable> EnterPsesAlcReflectionContext => s_enterPsesReflectionContextLazy.Value;
+
+        private readonly IServiceProvider _serviceProvider;
+
+        internal EditorEngine(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+            LanguageServer = new EditorLanguageServer(_serviceProvider.GetService<ILanguageServer>());
+        }
+
+        public EditorLanguageServer LanguageServer { get; }
+
+        public object GetService(string psesServiceFullTypeName) => GetService(psesServiceFullTypeName, "Microsoft.PowerShell.EditorServices");
```

Comment:
So this should allow you to access arbitrary services from within the PSES ALC

Reply:
Problem with this is that since the type that an external assembly would reference points to a different method table than the type returned by this method, you can't cast.  This might be useful for PowerShell based extension modules, but I think the normal PSES API is suitable enough for that scenario.